### DISCUSSION TITLE
How to handle unsaved work on file modified by an external process?

### DISCUSSION THREAD

**Discussion by iceghost** [helix-editor/helix#8468] [Q&A]
I want to ask how everyone does this. The only way I can do now is remember what I edited, reload, and redo what I remembered.

**Comment by the-mikedavis** [comment]
If you want to ignore the modification error and write what you have in Helix to the file anyways you can use `:write!`. You can also write the contents to some other temporary path (`:write /tmp/file.txt`), reload, and then compare the changes.

**Reply by gabydd** [reply]
And you can undo to your previous work `u` right after you reload if you decide you just want the previous stuff or need to do something with it

**Reply by iceghost** [reply]
> If you want to ignore the modification error and write what you have in Helix to the file anyways you can use `:write!`. You can also write the contents to some other temporary path (`:write /tmp/file.txt`), reload, and then compare the changes.

Would be cool if there is a command to keep both, using `diff3`. The idea is get three versions: mine (with our changes, the current buffer), older (with no changes, the latest written buffer), and yours (with external changes, the "on-disk buffer"), then perform a three-way merge.

Currently, to obtain the older version, you can't simply `:reload` because that will give you the yours version. I will have to undo until the [+] mark disappears.

**Comment by dariooddenino** [comment]
This right now is currently a big pain point for me. 

The real problem is that the warning message is not that visible, and a lot of time I'm wasting time because I didn't notice that helix is not saving my changes at all.

I wonder if there's a way to make that more visible (like changing the buffer's background?)

**Reply by anatoliyarkhipov** [reply]
Or ideally a way to just disable this behavior - haven't once I found myself grateful for the file not being saved due to an external modification, yet the frustration is always immense.

**Reply by kirawi** [reply]
As was mentioned above, you can just do `:w!`. Adding a config option for this feels overkill

**Reply by anatoliyarkhipov** [reply]
What you're effectively saying is "You can just train yourself to always use `:w!` instead of `:w` and have a muscle memory different from most of the users".

Maybe you could explain the horrific situations it helps to prevent? It's more a psychological issue, starting to slowly realize the file isn't being saved for the last 15 minutes, and maybe if I knew what the price is paid for, it would be easier to accept it.

**Reply by pascalkuthe** [reply]
I have lost countless hours of work with old versions of hjx where I had multiple different helix versions running and accidentl `:wq` in an old instance (while the new instance was already closed). I also accidently overwrite changes made by git (or other editors).

Our behavior matches vim and is generally the correct conservative option. Usually if yhe file changes externally that is with good reason sp not voding those changes without warning is a good idea.

This feature ws highly requested. The UI could be improved but you are the only person that ever advocated for disabling it so clearly your preference to void what is on disk without warning or option for recovery is unusual. We will not be disabling this feature.

**Reply by kirawi** [reply]
https://github.com/helix-editor/helix/issues/4423 could, however, address your desire by allowing you to alias `:w` for `:w!`

**Reply by dariooddenino** [reply]
Yeah, I agree that the current behavior is the correct one (aside from having the buffer update automatically on disk changes).

What's really needed imho is just a better UI feedback.
